### PR TITLE
Update comp matrix to include Kubernetes 1.23

### LINKED ISSUE #615: Add Kubernetes 1.23 to the test matrix
Similar to #378, Kubernetes 1.23 has just been released and we need to add it to the test and compatibility matrix.

https://github.com/kubernetes/kubernetes/releases/tag/v1.23.0

### PR DESCRIPTION
**Pull request by shree007**
Closes #615.

This also removes 1.20, to keep the compat matrix manageable.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -246,16 +246,16 @@ The OpenTelemetry Operator *might* work on versions outside of the given range,
...
-| v0.40.0                | v1.20 to v1.22       | 1.6.1                |
-| v0.39.0                | v1.20 to v1.22       | 1.6.1                |
-| v0.38.0                | v1.20 to v1.22       | 1.6.1                |
-| v0.37.1                | v1.20 to v1.22       | v1.4.0 to v1.6.1     |
-| v0.37.0                | v1.20 to v1.22       | v1.4.0 to v1.5.4     |
-| v0.36.0                | v1.20 to v1.22       | v1.4.0 to v1.5.4     |
-| v0.35.0                | v1.20 to v1.22       | v1.4.0 to v1.5.4     |
-| v0.34.0                | v1.20 to v1.22       | v1.4.0 to v1.5.4     |
-| v0.33.0                | v1.20 to v1.22       | v1.4.0 to v1.5.4     |
+| v0.41.0                | v1.21 to v1.23       | 1.6.1                |
```
Comment (jpkrohling) [inline README.md:249]: This is not correct. Released versions shouldn't be changed, the matrix is changed only from this point and on. So, just add a new entry like this:

```
| v0.42.0 (unreleased)               | v1.21 to v1.23       | 1.6.1                |
```

Comment (shree007) [reply README.md:249]: Noted.

Comment (shree007) [reply README.md:249]: @jpkrohling  Please check, I have updated

Comment (jpkrohling) [reply README.md:249]: 0.41.0 has been released already without this PR, so, it really has to be "0.42.0 (unreleased)" here.

Comment (shree007) [reply README.md:250]: @jpkrohling updated.

File: .github/workflows/e2e.yaml [hunk 2]
```diff
@@ -14,9 +14,9 @@ jobs:
       matrix:
        kube-version:
        - "1.19"
```
Comment (VineethReddy02) [inline .github/workflows/e2e.yaml:16]: @jpkrohling any specific reason, why are we dropping `1.20` but not the `1.19` ?

File: README.md (hunk 1 above)
Comment (jpkrohling) [reply README.md:250]: 0.41.0 is still wrong, it shouldn't be changed at all, as it has been released already.

File: .github/workflows/e2e.yaml (hunk 2 above)
Comment (jpkrohling) [reply .github/workflows/e2e.yaml:16]: I thought we had removed 1.19 already. It should indeed be removed.

File: README.md [hunk 3]
```diff
@@ -246,7 +246,8 @@ The OpenTelemetry Operator *might* work on versions outside of the given range,
 
 | OpenTelemetry Operator | Kubernetes           | Cert-Manager         |
 |------------------------|----------------------|----------------------|
-| v0.41.0                | v1.20 to v1.22       | 1.6.1                |
+| v0.42.0 (unreleased)   | v1.21 to v1.23       | 1.6.1
+| v0.41.0                | v1.21 to v1.22       | 1.6.1                |
```
Comment (jpkrohling) [inline README.md:250]: This should really be left untouched :-)

**Comment by VineethReddy02** [comment]
closing this PR in favor to #633

**Pull request closed without merging**